src/evict.c

```diff
@@ -256,6 +256,20 @@ void evictionPoolPopulate(int dbid, dict *sampledict, dict *keydict, struct evic
     }
 }
 
+/* Reset the eviction pool. */
+void evictionPoolReset(struct evictionPoolEntry *pool) {
+    int i;
+    for (i = 0; i < EVPOOL_SIZE; i++) {
+        if (pool[i].key == NULL) continue;
+
+        /* Remove the entry from the pool. */
+        if (pool[i].key != pool[i].cached)
```

Comment:
@soloestoy in line 525, there is the same code to free. Could you make it once?
What do you think?

https://github.com/antirez/redis/blob/0e086dedf1e9e004220b87a3844a7905bc9b0a59/src/evict.c#L525